spec/controllers/api/assignment_repos_controller_spec.rb

```diff
@@ -35,13 +35,34 @@
         expect(json.first["username"]).to eq(user.github_user.login)
       end
 
-      it "returns repo url" do
-        expect(json.first["repoUrl"]).to eq(@assignment_repo.github_repository.html_url)
-      end
-
       it "returns user display name" do
         expect(json.first["displayName"]).to eq(user.github_user.name)
       end
     end
   end
+
+  describe "GET #clone_url", :vcr do
+    before do
+      @assignment_repo = create(:assignment_repo, assignment: assignment, github_repo_id: 42, user: user)
+
+      get :clone_url, params: {
+        organization_id: organization.slug,
+        assignment_id: assignment.slug,
+        assignment_repo_id: @assignment_repo.id,
+        access_token: user.api_token
+      }
+    end
+
+    after do
+      AssignmentRepo.destroy_all
+    end
+
+    it "returns success" do
+      expect(response).to have_http_status(:success)
+    end
+
+    it "returns json with temp clone url" do
+      expect(json["temp_clone_url"]).to_not be_nil
```

Comment:
It would be nice to compare the entire json blob here if possible. That way we can easily look at the tests to see what the API endpoint returns.

Reply:
Great idea!! Chatted with @d12 about this too, and ended up doing a similar thing to what you suggested. Combined all of JSON checks into one test but still kept them as individual `expect` calls so we know which attribute is failing from the tests!